### PR TITLE
Add command for materializing columns

### PR DESCRIPTION
**Pull request by macobo**
Expecting this to get used in both dev as well as when improving
upstream installations

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: ee/management/commands/materialize_columns.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+class Command(BaseCommand):
+    help = "Materialize properties into columns in clickhouse"
+
+    def add_arguments(self, parser):
+        parser.add_argument("--dry-run", action="store_true", help="Print plan instead of executing it")
+
+        parser.add_argument("--property", help="Property to materialize. Skips analysis.")
+        parser.add_argument(
+            "--property-table", type=str, default="events", choices=["events", "person"], help="Table of --property"
```
Comment (hazzadous) [inline ee/management/commands/materialize_columns.py:23]: Would specifying this also mean that we wouldn't backfill? Is this desired?

We could instead pass in the property to `materialize_properties_task` and reuse the logic there.

**Review by hazzadous — CHANGES_REQUESTED** [review]
Minor points. I'm not sure what the desired backfill behaviour is, maybe just adding "skips analysis and backfill" ?

File: ee/management/commands/materialize_columns.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            "--analyze-period",
+            type=int,
+            default=MATERIALIZE_COLUMNS_ANALYSIS_PERIOD_HOURS,
+            help="How long of a time period to analyze. Same as MATERIALIZE_COLUMNS_ANALYSIS_PERIOD_HOURS env variable.",
+        )
+        parser.add_argument(
+            "--max-columns",
+            type=int,
+            default=MATERIALIZE_COLUMNS_MAX_AT_ONCE,
+            help="Max number of columns to materialize. Same as MATERIALIZE_COLUMNS_MAX_AT_ONCE env variable.",
```
[the remark this answers is not in the document]
Comment (macobo) [reply ee/management/commands/materialize_columns.py:48]: Exactly. Problem is that:
1. Materialized columns are not completely free cost-wise (storage + ingest)
2. Backfilling is expensive

Meaning that we limit the amount of things we materialize at once and prioritize to those with the largest effect.

This task is also run in a cronjob already, adding this command mainly for:
1. Simplifying dev
2. Giving more granular tools if an instance needs speeding up

File: ee/management/commands/materialize_columns.py (hunk 1 above)
Comment (macobo) [reply ee/management/commands/materialize_columns.py:23]: Resolved!

File: ee/management/commands/materialize_columns.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hazzadous) [reply ee/management/commands/materialize_columns.py:48]: I think max columns or column limit make sense. Just the env var that sounds a little off.

I’d say the context you’ve given would be useful to add, either just as a comment or in the help text. Renaming the vat seems out of scope

**Review by hazzadous — APPROVED** [review]

**Pull request merged**